### ISSUE TITLE
Additional scaling transforms

### ISSUE THREAD

**Issue by gjmulder** [awslabs/gluon-ts#463]
@rshyamsundar mentioned that the Box Cox lambda parameter was being fit as part of the GluonTS model fitting, which sounds like an elegant approach.

However, looking at:

https://github.com/awslabs/gluon-ts/blob/master/src/gluonts/block/scaler.py

It seems that by default GluonTS performs only mean scaling.

Any ideas if you are planning to add the Box-Cox self-fitting scaler (a bit beyond my current knowledge and ability...), or should I go ahead and directly apply some simple scaling transforms myself, e.g. using sklearn's scalers:

https://scikit-learn.org/stable/modules/classes.html#module-sklearn.preprocessing

**Comment by mbohlkeschneider** [comment]
Hi @gjmulder, 

I'm not sure whether a learnable version of box-cox is currently implemented (maybe @rshyamsundar  knows?), but you can find the box-cox transform here: https://github.com/awslabs/gluon-ts/blob/master/src/gluonts/distribution/box_cox_tranform.py

On a side note: Depending on what you want to do, you want to take the log determinant jacobian into account to scale the area under the probability density function correctly when you apply a transformation. For this purpuse, GluonTS implements [bijections](https://github.com/awslabs/gluon-ts/blob/master/src/gluonts/distribution/bijection.py).  The scaler you are referring to, is conceptually a bit tricky because it is conditioned on some conditioning range. Thus, it does not currently fit into the bijections scheme. 

You might also implement transformations as part of the transformation chain (for some transformations it is not so critical.

Hope that helped. Let me know if you have further questions.

**Comment by rshyamsundar** [comment]
The learnable version of box-cox is already implemented and can be invoked via `TransformedDistributionOutput` in case of DeepAR like this (modulo a simple validation fix #475):

```
estimator = DeepAREstimator(
            freq="5min", prediction_length=12,
            distr_output=distribution.TransformedDistributionOutput(
                distribution.GaussianOutput(), [distribution.InverseBoxCoxTransformOutput(lb_obs=-1.0E-5)]
            )
          )
```

The test that verifies Box Cox parameter is learned can be found [here](https://github.com/awslabs/gluon-ts/blob/ca8d14d28c1444958818cb2ba1ee17c3f2ecaf48/test/distribution/test_distribution_inference.py#L586).

`DeepStateEstimator` is hard-coded to have LDS distribution and cannot currently use BoxCox transformation; please add an issue in case you need it.

**Comment by gjmulder** [comment]
Thanks both @mbohlkeschneider @rshyamsundar for the feedback.

I'll need to spend some time getting my head around and experimenting with the techniques you're proposing...
